### PR TITLE
Enable ResolveProjectReferences target to run

### LINKED ISSUE #2824: MSBuildWorkspace can't compile Class Library referencing PCL
Related to #560 and various dupes #1373, #2700, #1894

We first noticed this problem when trying to compile [XUnit](https://github.com/xunit/xunit) with `MSBuildWorkspace`. All the Desktop projects failed to build but all the PCLs were fine.

To repro:

 - Create new ClassLibrary project (I've called mine Desktop)
 - Create new PortableClassLibrary project (I'm called mine PortableClassLibrary)
 - Create reference to PortableClassLibrary in ClassLibrary
 - The projects must interact in some way. For example:

In the ClassLibrary I use the following:

```CSharp
namespace Desktop
{
    public class Class1
    {
        public void MyMethod()
        {
            var test = new PortableClassLibrary.Class1();
            var obj = test.PCLMethod();
        }
    }
}
```

And in the PCL:

```CSharp
namespace PortableClassLibrary
{
    public class Class1
    {
        public object PCLMethod()
        {
            return new object();
        }
    }
}
```

Now in another solution try to compile the above with Roslyn:

```CSharp
var msws = MSBuildWorkspace.Create(new Dictionary<string, string> { { "CheckForSystemRuntimeDependency", "true" } });
var soln = msws.OpenSolutionAsync(@"C:\Users\Josh\Documents\GitHub\Orange\Desktop\Desktop.sln").Result;
var comp = soln.Projects.First().GetCompilationAsync().Result;
var errs = comp.GetDiagnostics().Where(n => n.Severity == DiagnosticSeverity.Error).ToList();
//One error
```

*Expected:* Project builds without errors (as it does in Visual Studio).

*Actual:* We receive the following error:
```
C:\Users\Josh\Documents\GitHub\Orange\Desktop\Desktop\Class1.cs(14,23): error CS0012: The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
```

### PR DESCRIPTION
**Pull request by mattwar**
Fixes #2824 

This change enables MSBuildWorkspace to open projects that reference the Portable Class Libraries.

The technique used to remove assemblies associated with project references from the list of metadata references for the project being opened was disabling the MSBuild target that also did some shenanigans with portable class library references. The fix is to use a post-MSBuild filter to remove unwanted references instead of disabling some of the MSBuild targets.

@Pilchie @jasonmalinowski  please review

### CODE REVIEW

**Comment by Pilchie** [comment]
Can we add a test for the issues in #2824?  Namely, that we managed to pull in facade references in that case?

Otherwise, the code change LGTM!

File: src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs [hunk 1]
```diff
@@ -182,6 +177,11 @@ protected virtual string GetAssemblyName()
             return PathUtilities.GetFileName(assemblyName);
         }
 
+        protected bool IsProjectReferenceOutputAssembly(MSB.Framework.ITaskItem item)
+        {
+            return item.GetMetadata("ReferenceOutputAssembly") == "true";
```
Comment (Pilchie) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: Would `item.GetMetadata("ReferenceSourceTarget") == "ProjectReference"` be better here?

Comment (Pilchie) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: Possibly that doesn't flow into the references.  An alternative here might be to look at the `_ResolvedProjectReferencePaths` item group instead and remove them from the rest of the references

Comment (jasonmalinowski) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: :+1: to @Pilchie's question -- he found the metadata name faster than I.

**Comment by mattwar** [comment]
@dotnet-bot test this please

**Comment by MattGertz** [comment]
Approved 6/11 by ML Shiproom.

**Comment by tmeschter** [comment]
@dotnet-bot Retest this, please.

**Pull request merged**